app/components/channel_list/categories/categories.tsx

```diff
@@ -20,23 +21,27 @@ const styles = StyleSheet.create({
     },
 });
 
-const renderCategory = (data: {item: CategoryModel}) => {
-    return (
-        <>
-            <CategoryHeader category={data.item}/>
-            <CategoryBody category={data.item}/>
-        </>
-    );
-};
-
 const Categories = (props: Props) => {
-    if (!props.categories.length) {
-        return <LoadCategoriesError/>;
-    }
+    const intl = useIntl();
+    const renderCategory = useCallback((data: {item: CategoryModel}) => {
```

Comment:
You can move this `renderCategory` method outside of the `Categories` component.  

The deps array is watching for `props.categories` whereas it should be watching for the `data.item` ...but moving it outside would solve it.

Reply:
Now, that's it using useCallback, it's fine.